### PR TITLE
Topic recognition, Task 1 segmentation of ISIC2018 with improved UNet

### PR DESCRIPTION
**Pull request by masaet**
# Task 1, segmentation of ISIC2018 with the improved UNet

Name: Marius Saether
Student id: 48242099

This pull request implements the improved UNet architecture, which trains to perform binary segmentation on the dermoscopy images found in the ISIC2018 dataset. 

The model achieved an average DCS score of 0.834 on the validation set, but struggles with two images below 0.01. 
Necessary information attached in README.md

### Files included:
* dataset.py: Program for converting ISIC2018 dataset into a train-, validation- and testset, and creating the dataloaders. 

* modules.py: Program containing the improved uNet module, and creating the model 

* train.py: Program for training, validating and testing the model

* predict.py: Can load a pretrained model, and run it on the validation set. Wil also plot DCS scores and save segmented images. 

* utils.py: Dice loss function, and transformation-module used on the dataset

### Dependancies
* Pytorch: 2.0.1
* cuda: 11.7
* python: 3.10.12
* scikit-learn: 1.3.0
* pillow: 9.4.0
* matplotlib: 3.7.1

### CODE REVIEW

**Comment by wangzhaomxy** [comment]
This is an initial inspection, no action is required at this point.

- Well organized files.
- Algorithm solves the problem appropriately. Average Dice Coefficient in testing dataset: 0.834. Predicted Mask output looks good. No data augmentation, just train/test data split.
- Correctly using PyTorch to construct the Improved U-Net models and functions.
- Good coding design. Good handwriting code.
- Sufficient docstring and comment in the code.
- Easy difficulty problem, no higher difficulty marks.

**Comment by shakes76** [comment]
# Marking
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate design and implementation
Good spacing and comments
Header blocks
## Recognition Problem 
Solves problem 
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage
Module present
Commenting
No Data leakage, no augmentation -1
Difficulty: Easy -10
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
ReadMe acceptable, no usage -1
Model/technical explanation
Good Description and Comments 
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
No Feedback required
Request Description good
